File: MyMonitorHub.Agent/MyMonitorHub.Agent.Plugins/Plugins/CommandWatch.cs
using System;
using MyMonitorHub.Common.WebApi;
using MyMonitorHub.Agent.Client;
using MyMonitorHub.Agent.Core;
using Serilog;

namespace MyMonitorHub.Agent.Plugins
{
    public class CommandWatch : AbstractPlugin
    {
        private static readonly ILogger Logger = Log.ForContext<CommandWatch>();

        private const string Category = "CMD PRC";
        private const string SubCategory = "Thread";
        private const string Name = "Ping";

        protected override void Execute()
        {
            var client = new EventBroadcastClient(Core.Monitor.Port);
            try
            {
                client.Ping(); // now throws an error instead of supressing it
                Logger.Debug("ok");

                SayStatus(Category, SubCategory, Name, "OK", EventType.Ok); // should not throw an error
            }
            catch (Exception e)
            {
                HandleFail(e);
            }
        }

        private void HandleFail(Exception e)
        {
            var ex = e.InnerException ?? e;

            var message = ex.Message;
            if (string.IsNullOrEmpty(message))
            {
                message = ex.ToString();
            }
            SayStatus(Category, SubCategory, Name, message, EventType.Fail);
            Logger.Error(ex, "CommandWatch execution failed");
        }
    }
}

File: MyMonitorHub.JavaAgent/src/main/java/mymonitorhub/agent/plugins/CommandWatch.java
package mymonitorhub.agent.plugins;

import mymonitorhub.agent.client.EventBroadcastClient;
import mymonitorhub.agent.core.AbstractPlugin;
import mymonitorhub.agent.core.Monitor;
import mymonitorhub.agent.model.EventType;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Verifies the command listener thread is alive by sending a local MCM ping.
 * Mirrors C# CommandWatch.
 */
public class CommandWatch extends AbstractPlugin {

    private static final Logger log = LoggerFactory.getLogger(CommandWatch.class);

    private static final String CATEGORY     = "CMD PRC";
    private static final String SUB_CATEGORY = "Thread";
    private static final String NAME         = "Ping";

    @Override
    protected void execute() {
        EventBroadcastClient client = new EventBroadcastClient(Monitor.getPort());
        try {
            client.ping();
            log.debug("ok");
            sayStatus(CATEGORY, SUB_CATEGORY, NAME, "OK", EventType.Ok);
        } catch (Exception e) {
            handleFail(e);
        }
    }

    private void handleFail(Exception e) {
        Throwable cause = e.getCause() != null ? e.getCause() : e;
        String message = cause.getMessage();
        if (message == null || message.isEmpty()) {
            message = cause.toString();
        }
        sayStatus(CATEGORY, SUB_CATEGORY, NAME, message, EventType.Fail);
        log.error("CommandWatch execution failed: {}", message);
    }
}
